examples/core-hello-world/src/main/scala/com/mesosphere/usi/helloworld/CoreHelloWorld.scala

```diff
@@ -0,0 +1,166 @@
+package com.mesosphere.usi.helloworld
+import java.util.UUID
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{Flow, Keep, Sink, Source}
+import com.mesosphere.mesos.client.MesosClient
+import com.mesosphere.mesos.conf.MesosClientSettings
+import com.mesosphere.usi.core.Scheduler
+import com.mesosphere.usi.core.matching.ScalarResourceRequirement
+import com.mesosphere.usi.core.models.{Goal, PodId, PodSpec, PodStatus, PodStatusUpdated, ResourceType, RunSpec, SpecUpdated, SpecsSnapshot, StateEvent, StateSnapshot}
+import com.typesafe.config.{Config, ConfigFactory}
+import com.typesafe.scalalogging.StrictLogging
+import org.apache.mesos.v1.Protos.{FrameworkInfo, TaskState, TaskStatus}
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+import scala.sys.SystemProperties
+import scala.util.{Failure, Success}
+
+/**
+  * Run the hello-world example framework that:
+  *  - relies on our `core` module to handle Mesos connection, event handling, offer matching etc. as
+  *    opposed to the `simple-hello-world` framework which uses only the Mesos client and has to implement
+  *    all of the above mentioned pieces itself
+  *  - starts one `echo "Hello, world" && sleep N` task
+  *  - exits should the task fail (or fail to start)
+  *
+  *  Good to test against local Mesos.
+  *
+  */
+class CoreHelloWorld(conf: Config) extends StrictLogging {
+  implicit val system = ActorSystem()
+  implicit val mat = ActorMaterializer()
+  implicit val ec = system.dispatcher
+
+  val settings = MesosClientSettings(conf.getString("master-url"))
+  val frameworkInfo = FrameworkInfo
+    .newBuilder()
+    .setUser(
+      new SystemProperties()
+        .get("user.name")
+        .getOrElse(throw new IllegalArgumentException("A local user is needed to launch Mesos tasks")))
+    .setName("CoreHelloWorldExample")
+    .setFailoverTimeout(0d)
+    .build()
+
+  val client: MesosClient = Await.result(MesosClient(settings, frameworkInfo).runWith(Sink.head), 10.seconds)
```

Comment:
Hm, I wonder if we really should provide a constructed client or if we should just inject an implementation. What do you think, @timcharper?